operators/text/string_join.cc

```diff
@@ -16,6 +16,8 @@ void KernelStringJoin::Compute(OrtKernelContext* context) {
   std::vector<std::string> X, sep;
   GetTensorMutableDataString(api_, ort_, context, input_X, X);
   GetTensorMutableDataString(api_, ort_, context, input_sep, sep);
+  if (X.size() == 0)
```

Comment:
@TomWildenhain-Microsoft : That's perfect!  Two questions:
1) When is your explanation going to be added to the ONNX standard?
2) Is there already a single-line check in the onnxruntime-extensions repo ( or in the onnxruntime repo ) that we can use in all applicable operators to ensure zero-dimensional tensors with no values will end with ORT_INVALID_ARGUMENT?

Reply:
@joburkho It's possible that input dimension could be [] which is scalar, and following check 
  if (*axis < 0 || *axis >= dimensions.size()) 
seems breaking the case.

X.size() > 0 seems redundant, or if there is some cases I don't know @Zuwei-Zhao

Reply:
> 1. scalar support (dim = []): dimensions.size() will be zero, so *axis < 0 || *axis >= dimensions.size() will be always be true, that's the thing we should fix as well.

Sounds good.

> 2. The access violation(dim = [0]), either we can support it (taking it as an empty string). or it should be filtered out by the caller. Throwing an exception here seems not be very helpful because if it is non-exception compiling, it still causes a crash (abort).

This code is being used both in places where we have exceptions turned off and in places where we have exceptions turned on.  In the places where we have exceptions turned on, an access violation is not acceptable ... only an exception is acceptable.

Reply:
How about the non-exception case? that seems to more related to production.

Thing becomes a little complicated in the custom op scenario since its doesn't have a strict schema, otherwise ONNXRuntime will check the input and throw an exception if necessary.

Instead of throwing an exception here, if it zero dim input encounters, the output can be the same as the input.